client/lib/posts/post-list-cache-store.js

```diff
@@ -34,6 +35,10 @@ function getCacheKey( options ) {
 }
 
 function get( query ) {
+	if ( config.isEnabled( 'sync-handler' ) ) {
+		return console.warn( 'synchandler is enabled. Disable post-list-cache-store' );
```

Comment:
@retrofox I notice a significant cost to speed of rendering the posts when we disable post-list-cache-store. See this video... the first page is using post-list-cache-store, and switching between different post-lists is instant, and the second example is showing this branch, without post-list-cache-store. https://cloudup.com/cJyfbAowH53

So I don't think we should skip this functionality, since it's much faster to render the Posts that are in memory, rather than the ones that are cached via indexedDB. However, we should make sure we run a query to update these posts as well. I think my PR here (https://github.com/Automattic/wp-calypso/pull/2965) would take care of this, so I'd like to get that merged, and then have your PR rebased onto it. But if we can't have the same speed without post-list-cache-store, then I would not remove it.